### PR TITLE
Allow users to make S3 deletion optional

### PR DESCRIPTION
**Pull request by yunjiang29**
https://issues.redhat.com/browse/CORS-2598 Make S3 deletion optional for AWS

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-preserve-ign-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-preserve-ign-f28

**Comment by yunjiang29** [comment]
@gpei @jianlinliu PTAL, thanks

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-ref.yaml [hunk 1]
```diff
@@ -77,6 +77,12 @@ ref:
       * "" (default) - corresponds to "Optional" behaviour. 
       * "Required" - sets IMDSv2 to Required for all machines in install-config, metadata service is not accessible without auth procedure.
       * "Optional" - auth procedure on metadata service is optional
-
+  - name: PRESERVE_BOOTSTRAP_IGNITION
+    default: ""
+    documentation: |-
+      Allow users to make S3 deletion optional
+      More info:
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-ref.yaml:84]: Maybe we can drop the "More info" to be more tidy.

File: ci-operator/step-registry/cucushift/installer/check/aws/bootstrap-ignition/cucushift-installer-check-aws-bootstrap-ignition-commands.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+set -o pipefail
+
+REGION="${LEASED_RESOURCE}"
+INFRA_ID=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+
+bucket_name=${INFRA_ID}-bootstrap
+key_name="bootstrap.ign"
+
+echo "Checking if s3://${bucket_name}/${key_name} exists"
+aws --region ${REGION} s3 ls s3://${bucket_name}/${key_name}
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/aws/bootstrap-ignition/cucushift-installer-check-aws-bootstrap-ignition-commands.sh:14]: Maybe after the check succeeded, we can clean the S3 bucket in this step?

Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/check/aws/bootstrap-ignition/cucushift-installer-check-aws-bootstrap-ignition-commands.sh:14]: The bucket will be removed by `destroy` command, any concerns about this?

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/aws/bootstrap-ignition/cucushift-installer-check-aws-bootstrap-ignition-commands.sh:14]: Ah, I thought installer would be destroy the preserved bootstrap ignition, if that, pls ignore this comment.

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-ref.yaml (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-ref.yaml]: Updated.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-preserve-ign-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-preserve-ign-f28

File: ci-operator/step-registry/cucushift/installer/check/aws/bootstrap-ignition/cucushift-installer-check-aws-bootstrap-ignition-commands.sh (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/aws/bootstrap-ignition/cucushift-installer-check-aws-bootstrap-ignition-commands.sh:14]: This command need aws credential.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-preserve-ign-f28

**Comment by yunjiang29** [comment]
Test:
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/41764/rehearse-41764-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-preserve-ign-f28/1701232874511929344/artifacts/aws-ipi-preserve-ign-f28/cucushift-installer-check-aws-bootstrap-ignition/build-log.txt

```
Checking if s3://ci-op-57tdwvyr-5b15b-8hb4s-bootstrap/bootstrap.ign exists
2023-09-11 13:59:11     346958 bootstrap.ign
```

**Comment by jianlinliu** [comment]
cc @liangxia to review

**Pull request merged**